### PR TITLE
Add HIP Backend test support for attention FP16

### PR DESCRIPTION
**Pull request by erman-gurses**
Add HIP Backend test support for attention FP16.
It is based on this test suite: https://github.com/iree-org/iree/pull/18320

### CODE REVIEW

File: tests/e2e/attention/CMakeLists.txt [hunk 1]
```diff
@@ -78,3 +78,96 @@ iree_generated_e2e_runner_test(
...
+    "--query_type=f16"
+    "--key_type=f16"
+    "--value_type=f16"
+    "--shapes=small"
+  TEST_RUNNER
+    iree_tools_testing_e2e_iree-e2e-attention-test
+  TARGET_BACKENDS
+    "rocm"
+  DRIVERS
+    "hip"
```
Comment (ScottTodd) [inline tests/e2e/attention/CMakeLists.txt:107]: Do Vulkan, CUDA, and Metal also work?

We shouldn't be testing on only one GPU backend.

Comment (erman-gurses) [reply tests/e2e/attention/CMakeLists.txt:107]: Sounds good; I can add them, too.

Comment (erman-gurses) [reply tests/e2e/attention/CMakeLists.txt:107]: Per our conversation, the other backend tests will be added later.

Comment (ScottTodd) [reply tests/e2e/attention/CMakeLists.txt:107]: Vulkan logs for future reference: https://github.com/iree-org/iree/actions/runs/10562819042/job/29261656512#step:4:9893

```
[410/1578] Generating /work/full-build-dir/tests/e2e/attention/e2e_attention_gpu_vulkan_f16_f16_f16_large_vulkan-spirv_vulkan_attention.vmfb from e2e_attention_gpu_vulkan_f16_f16_f16_large_vulkan-spirv_vulkan_attention.mlir
FAILED: tests/e2e/attention/e2e_attention_gpu_vulkan_f16_f16_f16_large_vulkan-spirv_vulkan_attention.vmfb /work/full-build-dir/tests/e2e/attention/e2e_attention_gpu_vulkan_f16_f16_f16_large_vulkan-spirv_vulkan_attention.vmfb 
cd /work/full-build-dir/tests/e2e/attention && /work/full-build-dir/tools/iree-compile --output-format=vm-bytecode --mlir-print-op-on-diagnostic=false --iree-hal-target-backends=vulkan-spirv --iree-vulkan-target=rdna3 /work/full-build-dir/tests/e2e/attention/e2e_attention_gpu_vulkan_f16_f16_f16_large_vulkan-spirv_vulkan_attention.mlir -o /work/full-build-dir/tests/e2e/attention/e2e_attention_gpu_vulkan_f16_f16_f16_large_vulkan-spirv_vulkan_attention.vmfb --iree-hal-executable-object-search-path=\"/work/full-build-dir\"
/work/full-build-dir/tests/e2e/attention/e2e_attention_gpu_vulkan_f16_f16_f16_large_vulkan-spirv_vulkan_attention.mlir:4:14: error: failed to run translation of source executable to target executable for backend #hal.executable.target<"vulkan-spirv", "vulkan-spirv-fb", {iree.gpu.target = #iree_gpu.target<arch = "rdna3", features = "spirv:v1.6,cap:Shader", wgp = <compute =  fp64|fp32|fp16|int64|int32|int16|int8, storage =  b64|b32|b16|b8, subgroup =  shuffle|arithmetic, dot =  dp4xi8toi32, mma = [<WMMA_F32_16x16x16_F16>, <WMMA_F16_16x16x16_F16>], subgroup_size_choices = [32, 64], max_workgroup_sizes = [1024, 1024, 1024], max_thread_count_per_workgroup = 1024, max_workgroup_memory_bytes = 65536, max_workgroup_counts = [2147483647, 2147483647, 2147483647]>>}>
  %result1 = iree_linalg_ext.attention {
             ^
/work/full-build-dir/tests/e2e/attention/e2e_attention_gpu_vulkan_f16_f16_f16_large_vulkan-spirv_vulkan_attention.mlir:1:1: note: called from
func.func @attention_2_1024_256_128_64_dtype_f16_f16_f16_f16(%query: tensor<2x1024x256xf16>, %key: tensor<2x128x256xf16>, %value: tensor<2x128x64xf16>, %scale: f32) -> tensor<2x1024x64xf16> {
^
failed to translate executables

**Review by ScottTodd — APPROVED** [review]

**Pull request merged**